### ISSUE TITLE
Experiments table is broken on a branch w/o history?

### ISSUE THREAD

**Issue by shcheklein** [iterative/vscode-dvc#2703]
Running on this branch https://github.com/shcheklein/ensemble-dvc-template/tree/experiment

I'm getting an empty table:

<img width="1512" alt="Screen Shot 2022-10-30 at 1 10 52 PM" src="https://user-images.githubusercontent.com/3659196/198899520-f1c9a3cd-6369-483f-86c1-015464a5fba4.png">

While `dvc exp show` is working properly:

It's a critical issue if it's broken on any branch like this.

Extension version: v0.5.7
DVC version: 2.30.0

**Comment by julieg18** [comment]
I cloned the repo and checked out to the `experiment` branch, but I'm seeing the table: 

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/43496356/199026505-26a12e4f-0639-47be-b7be-e3c03d0c2426.png">

Is the DVC output or VSCode dev tools showing any errors?

**Comment by shcheklein** [comment]
I can reproduce it with a fresh clone.

(had to update `Live.log` to `Live.log_metric` since API has changed)


https://user-images.githubusercontent.com/3659196/199155928-6e67a1cf-afaa-40ee-901e-66003d014a00.mov

**Comment by julieg18** [comment]
> I can reproduce it with a fresh clone.

Thanks for the example video! I managed to reproduce it on this repo and in a starter project. Will shift my focus to debugging this!

**Comment by julieg18** [comment]
From my testing so far, the `dvc exp show --num` command is broken on branches without history when you create a branch with `dvc branch`. 

https://user-images.githubusercontent.com/43496356/199589691-26fed2b0-11ac-4c10-833b-032ca0495415.mov

Steps to reproduce
```bash
dvc exp branch EXP_NAME BRANCH_NAME
git checkout BRANCH_NAME
dvc exp show -n 3
```

cc @shcheklein, @pmrowla, @karajan1001

**Comment by blakeNaccarato** [comment]
I've got a very minimal reproduction of this issue, triggered seemingly by a JSON metrics file containing any `NaN`. I present two repos, one where experiments load properly in the DVC sidebar of the VSCode extension, and one where experiments DO NOT load in the DVC sidebar, as in the original post above.

In both cases, `dvc exp show` works, it's just the extension which fails when any `NaN` are found in a metrics file.

My DVC VSCode extension is v0.5.10
My DVC CLI version is 2.33.2
I am using Python 3.10.6 on Windows 10.
Output of `platform.version()` is `'Windows-10-10.0.19044-SP0'`
Output of `platform.architecture()` is `('64bit', 'WindowsPE')`

Both repos have the following `dvc.yaml`:

```
stages:
  test:
    cmd: "python test.py"
    metrics:
      - "out.json"
```

The file `test.py` simply takes the contents of `in.json` and writes it to `out.json`.

The working example is in https://github.com/blakeNaccarato/dvc-vscode-2703-repro-nonan. Experiments DO LOAD properly in this repo. `in.json` contains the following:

```json
{
  "test": 1
}
```

The bug reproduction is over in https://github.com/blakeNaccarato/dvc-vscode-2703-repro-nan. Experiments DO NOT LOAD properly in this repo. `in.json` contains the following:

```json
{
  "test": NaN
}
```

**Comment by blakeNaccarato** [comment]
@shcheklein I've merged your pull request in the issue repro. Does it make a difference mechanically whether `in.json` was registered under `params` or `deps` for the stage? Since it's just a toy example to scare out the bug, I don't know whether that matters.

**Comment by blakeNaccarato** [comment]
I've added @shcheklein as a collaborator to that issue repro so you can push to main if needed. I'll If you need a Windows machine to reproduce Windows-specific stuff, I'm sure you could demonstrate that via a GitHub Action.

**Comment by shcheklein** [comment]
@blakeNaccarato I've create a separate issue #2733 . It's a different one. Let's keep the discussion there. Thanks for the report.

**Comment by julieg18** [comment]
The problem is fixed with https://github.com/iterative/dvc/pull/8541 merged and DVC's latest version, `2.34.1`. Do we need to update the extension's `LATEST_TESTED_CLI_VERSION` before closing this?

**Comment by shcheklein** [comment]
@julieg18 yes, I think we can close this if it's fixed. Bumping is a good idea (we should be doing this regularly independent of specific bugs, etc).
